src/PhpEnvironment/Response.php

```diff
@@ -82,7 +83,7 @@ public function contentSent()
     public function sendHeaders()
     {
         if ($this->headersSent()) {
-            return $this;
+            throw new RuntimeException('Cannot send headers, headers already sent');
```

Comment:
I tend to agree with @Ocramius that this would be a pretty egregious BC break.

One possibility that would resolve the issue as well as keep BC: we could add a "headers sent handler" callback to the response, and have it triggered here if it is present. Something like:

```php
private $headersSentHandler;

public function setHeadersSentHandler(?callable $handler): void
{
    $this->headersSentHandler = $handler;
}

public function sendHeaders()
{
    if ($this->headersSent()) {
        if ($this->headersSentHandler) {
            ($this->headersSentHandler)($this);
        }
        return;
    }

    // ...
```

If a user wants to throw an exception, they can either (a) add a delegator to the `Response` service, or (b - preferred) add a bootstrap listener that injects it:

```php
public function onBootstrap(MvcEvent $e): void
{
    $response = $e->getResponse();
    $response->setHeadersSentHandler(function ($response): void {
        throw new RuntimeException('Cannot send headers, headers already sent');
    });
}
```

This makes it opt-in, and, further, makes it easy for somebody to create a distributable module that enables the throw.

Reply:
@weierophinney implemented.